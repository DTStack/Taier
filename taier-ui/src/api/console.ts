/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * 系统管理
 */
import http from './http';
import req from './reqConsole';

export default {
	addTenant(params: any) {
		return http.post(req.ADD_TENANT, params);
	},
	switchTenant(params: { tenantId: number }) {
		return http.post(req.SWITCH_TENANT, params);
	},
	login(params: { username: string; password: string }) {
		return http.postAsFormData(req.LOGIN, params);
	},
	// 4.0 版本相关接口
	addCluster(params: { clusterName: string }) {
		return http.post(req.ADD_CLUSTER, params); // 新增集群
	},
	getClusterInfo(params: { clusterId: number }) {
		return http.get(req.GET_CLUSTER_INFO, params);
	},
	uploadResource(params: { fileName: any; componentType: number }) {
		return http.postAsFormData(req.UPLOAD_RESOURCE, params);
	},
	deleteComponent(params: { componentId: number }) {
		return http.post(req.DELETE_COMPONENT, params); // 删除组件
	},
	deleteCluster(params: { clusterId: number }) {
		return http.post(req.DELETE_CLUSTER, params);
	},
	testConnect(params: {
		clusterName: string;
		componentType: number;
		versionName: string;
		deployType: number | string;
	}) {
		return http.post(req.TEST_CONNECT, params);
	},
	testConnects(params: { clusterName: string }) {
		return http.post(req.TEST_CONNECTS, params);
	},
	closeKerberos(params: { componentId: number }) {
		return http.post(req.CLOSE_KERBEROS, params);
	},
	getVersionData(params?: any) {
		return http.get(req.GET_VERSION, params);
	},
	saveComponent(params: any) {
		return http.postAsFormData(req.SAVE_COMPONENT, params);
	},
	parseKerberos(params: any) {
		return http.postAsFormData(req.PARSE_KERBEROS, params);
	},
	getCompVersion(params: any) {
		return http.get(req.GET_COMPONENT_VERSION, params);
	},

	getResourceList(params: any) {
		return http.post(req.GET_RESOURCE_LIST, params);
	},
	getClusterList(params: { currentPage: number, pageSize: number, }) {
		return http.post(req.GET_CLUSTER_LIST, params);
	},
	uploadClusterResource(params: any) {
		return http.postAsFormData(req.UPLOAD_CLUSTER_RESOURCE, params);
	},
	getTenantList(params?: any) {
		return http.get(req.GET_TENANT_LIST, params);
	},
	getMetaComponent(params?: any) {
		return http.get(req.GET_META_COMPONENT, params);
	},
	testCluster(params: any) {
		return http.post(req.TEST_CLUSTER_CONNECT, params);
	},
	createCluster(params: any) {
		return http.postAsFormData(req.NEW_CLUSTER, params);
	},
	bindUserToQuere(params: any) {
		return http.post(req.BIND_USER_TO_RESOURCE, params);
	},
	updateCluster(params: any) {
		return http.postAsFormData(req.UPDATE_CLUSTER, params);
	},
	getQueueLists(params: any) {
		return http.post(req.GET_QUEUE_LISTS, params);
	},
	confirmSwitchQueue(params: any) {
		return http.post(req.CONFIRM_SWITCH_QUEUE, params);
	},
	// 4.1版本
	// 获取存储组件列表
	getComponentStore(params: any) {
		return http.post(req.GET_COMPONENTSTORE, params);
	},

	// 上传kerberos文件
	uploadKerberos(params: { kerberosFile: any; clusterId: string; componentCode: number }) {
		return http.postAsFormData(req.UPLOAD_KERBEROS, params);
	},

	// 更新krb5.conf文件
	updateKrb5Conf(params: { krb5Content: string }) {
		return http.post(req.UPDATE_KRB5CONF, params);
	},
	// 4.2版本
	getDtScriptAgentLabel(params: { agentAddress: string }) {
		return http.post(req.GET_DTSCRIPT_AGENT_LABEL, params);
	},
	getClusterComponentUser(params: {
		clusterId: string;
		needRefresh: boolean;
		componentTypeCode: number;
		agentAddress: string;
	}) {
		return http.post(req.GET_CLUSTER_COMPONENT_USER, params);
	},
	addOrUpdateComponentUser(params: { componentUserList: any[] }) {
		return http.post(req.ADD_OR_UPDATE_COMPONENT_USER, params);
	},
	// 任务管理模块
	// 概览-获取集群
	getClusterDetail(params: any) {
		return http.post(req.GET_CLUSTER_DETAIL, params);
	},
	// 概览-集群下拉列表
	getClusterSelect(params: any) {
		return http.post(req.GET_CLUSTER_SELECT, params);
	},
	// 获取节点下拉
	getNodeAddressSelect(params?: any) {
		return http.post(req.GET_NODEADDRESS_SELECT, params);
	},
	// 根据节点搜索

	// 明细-根据任务名搜索任务
	searchTaskList(params: any) {
		return http.post(req.SEARCH_TASKNAME_LIST, params);
	},
	// 明细-模糊查询任务名
	searchTaskFuzzy(params: any) {
		return http.post(req.SEARCH_TASKNAME_FUZZY, params);
	},
	// 明细-杀死选中或者杀死全部任务
	killTasks(params: {
		jobResource: string;
		nodeAddress: string;
		stage: number;
		jobIdList?: any[];
	}) {
		return http.post(req.KILL_TASKS, params);
	},

	killAllTask(params: { jobResource: string; nodeAddress: string }) {
		return http.post(req.KILL_ALL_TASK, params);
	},

	stickJob(params: { jobId: string; jobResource: string }) {
		return http.post(req.JOB_STICK, params);
	},
	// 引擎列表
	getEngineList(params?: any) {
		return http.post(req.GET_ENGINE_LIST, params);
	},
	// group列表
	getGroupList(params: any) {
		return http.post(req.GET_GROUP_LIST, params);
	},
	// 查看明细 和搜索条件
	getViewDetail(params: {
		stage: number;
		jobResource: string;
		nodeAddress?: string;
		currentPage: number;
		pageSize: number;
	}) {
		return http.post(req.GET_VIEW_DETAIL, params);
	},
	// 顺序调整调整优先级
	changeJobPriority(params: any) {
		return http.post(req.CHANGE_JOB_PRIORITY, params);
	},
	// 查看剩余资源
	getClusterResources(params: any) {
		return http.post(req.GET_CLUSTER_RESOURCES, params);
	},
	getLoadTemplate(params: any) {
		return http.post(req.GET_LOADTEMPLATE, params);
	},
	uploadKerberosFile(params: any) {
		return http.postAsFormData(req.UPLOAD_KERBEROSFILE, params);
	},
	getKerberosFile(params: any) {
		return http.post(req.GET_KERBEROSFILE, params);
	},
	testComponent(params: any) {
		return http.post(req.TEST_COMPONENT_CONNECT, params);
	},
	testComponentKerberos(params: any) {
		return http.postAsFormData(req.TEST_COMPONENT_CONNECT_KERBEROS, params);
	},
	addComponent(params: any) {
		return http.post(req.ADD_COMPONENT, params);
	},
	saveComponentWithKerberos(params: any) {
		return http.postAsFormData(req.SAVE_COMPONENT_KERBEROS, params);
	},
	deleteKerberos(params: any) {
		return http.post(req.DELETE_KERBEROS, params); // 删除Haddop Kerberos认证文件
	},
	addEngine(params: any) {
		return http.post(req.ADD_ENGINE, params);
	},
	addEngines(params: any) {
		return http.post(req.ADD_ENGINS, params);
	},
	updateClusterVersion(params: {
		clusterId: number;
		hadoopVersion: string;
		syncType: number /* 同步元数据组件类型 */;
	}) {
		return http.post(req.UPDATE_CLUSTER_VERSION, params);
	},
	// 资源管理
	getAllCluster(params?: any) {
		return http.get(req.GET_ALL_CLUSTER, params); // 返回数据包含集群下的engine，以及队列
	},
	getEnginesByCluster(params?: any) {
		return http.get(req.GET_ENGINES_BY_CLUSTER, params);
	},
	searchTenant(params: any) {
		return http.post(req.SEARCH_TENANT, params);
	},
	getTaskResourceTemplate(params?: any) {
		return http.post(req.TASK_RESOURCE, params);
	},

	getQueue(params: any) {
		return http.post(req.GET_QUEUE, params);
	},
	bindTenant(params: any) {
		return http.post(req.BIND_TENANT, params);
	},
	switchQueue(params: any) {
		return http.post(req.SWITCH_QUEUE, params);
	},
	bindNamespace(params: any) {
		return http.post(req.BIND_NAME_SPACE, params);
	},
	queryTaskResourceLimits(params: any) {
		return http.post(req.GET_TASKLIMITS, params);
	},

	/**
	 * 告警通道
	 */
	addOrUpdateAlarmRule(params: {
		alarmType: number;
		alertGateCode: number;
		alertGateSource: string;
		alertGateName: string;
		alertGateJson: string;
		alertTemplate: string;
		receiveMethod?: string;
		defaultChannel?: boolean;
		file?: any;
	}) {
		return http.postAsFormData(req.ADD_OR_UPDATE_ALARM, params);
	},
	getAlarmRuleList(params: {
		currentPage: number;
		pageSize: number;
		alertGateType?: any[];
		clusterId?: string;
	}) {
		return http.post(req.GET_ALARM_RULE_LIST, params);
	},
	deleteAlarmRule(params: { id: number }) {
		return http.post(req.DEL_ALARM_RULE_LIST, params);
	},
	getByAlertId(params: { id: number }) {
		return http.post(req.GET_ALARM_BY_ID, params);
	},
	setDefaultAlert(params: { alertId: number; alertGateType: number }) {
		return http.post(req.SET_DEFAULT_ALERT, params);
	},
	testAlert(params: any) {
		return http.postAsFormData(req.TEST_ALERT, params);
	},
	refreshQueue(params: { clusterName: string }) {
		return http.post(req.REFRESH_QUEUE, params);
	},
	getAlarmConfig() {
		return http.post(req.GET_ALARM_CONFIG);
	},
	updateAlarmConfig(params: { componentConfig: string; componentTemplate: string }) {
		return http.post(req.UPDATE_ALARM_CONFIG, params);
	},
	testAlarmConfig() {
		return http.post(req.TEST_ALARM_CONFIG);
	},
};
