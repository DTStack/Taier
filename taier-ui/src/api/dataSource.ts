/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { queryParse } from '@/utils';
import Swagger from './dataSources-swagger';
import http from './http';

type APIProps = Record<
	keyof typeof Swagger,
	(params: any, config?: any) => Promise<any>
>;

function mapUrlObjToFuncObj(urlObj: typeof Swagger) {
	const keys = Object.keys(urlObj) as (keyof typeof Swagger)[];
	return keys.reduce((pre, cur) => {
		const item = urlObj[cur];
		pre[cur] = async function (params, config: any = {}) {
			const queryParams: any = queryParse(window.location.hash);
			// 所有的请求接口统一带上header
			if (queryParams.datasource_id) {
				config.headers = new Headers({
					dt_datasource_id: queryParams.datasource_id,
				});
			}
			return await http[item.method as 'get' | 'post' | 'postForm'](
				item.url,
				params,
				config,
			);
		};
		return pre;
	}, {} as APIProps);
}
// function mapUrlObjToStrObj(urlObj) {
//   const Url = {};
//   keys(urlObj).forEach((key) => {
//     const item = urlObj[key];
//     Url[key] = item.url;
//   });
//   return Url;
// }

export const API = mapUrlObjToFuncObj(Object.assign({}, Swagger));
// export const URL = mapUrlObjToStrObj(Object.assign({}, Swagger, Restful));
