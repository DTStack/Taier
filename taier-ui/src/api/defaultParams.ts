import { cloneDeep, isEmpty, isObject } from "lodash";

export const convertRequestParams = (template = {}, params = {}) => {
    const loop = (template: any = {}, params: any = {}) => {
        const newData: any = {}
        Object.keys(template).forEach(k => {
            if (params[k] === undefined) {
                newData[k] = undefined;
                return;
            }
            const el = template[k];
            const d = cloneDeep(params[k]);
            if (isObject(el) && !isEmpty(el) && isObject(d)) {
                newData[k] = loop(el, d);
            } else {
                newData[k] = d
            }
        })
        return newData
    }
    return loop(template, params);
}
export const restfulDataPreview = {
    bodyStr: '',
    decode: '',
    fieldDelimiter: '',
    fields: '',
    headerStr: '',
    paramStr: '',
    requestInterval: 0,
    requestMode: '',
    url: '',
    protocol: ''
}

export const taskParams = {
    additionalResourceIdList: [],
    componentVersion: '',
    computeType: 0,
    createModel: 0,
    engineType: 0,
    exeArgs: '',
    forceUpdate: false,
    graphData: [],
    id: 1,
    isDeleted: 0,
    isDirtyDataManage: true,
    lockVersion: 1,
    mainClass: '',
    name: '',
    nodePid: 1,
    operatorJsons: '',
    preSave: false,
    readWriteLockVO: {
        getLock: false,
        gmtCreate: 0,
        gmtModified: 0,
        id: 0,
        isDeleted: 0,
        lastKeepLockUserName: 1,
        lockName: '',
        modifyUserId: 1,
        projectId: 1,
        relationId: 1,
        result: 1,
        type: '',
        version: 1
    },
    resourceIdList: [],
    settingMap: {},
    side: [],
    sideParams: '',
    sink: [],
    sinkParams: '',
    source: [],
    sourceList: [],
    sourceMap: {},
    sourceParams: '',
    sqlText: '',
    strategyId: 1,
    streamTaskDirtyDataManageVO: {
        id: 0,
        linkInfo: {},
        logPrintInterval: 0,
        maxCollectFailedRows: 0,
        maxRows: 0,
        outputType: '',
        taskId: 0
    },
    scheduleConf: '',
    targetMap: {},
    taskDesc: '',
    taskId: 1,
    taskParams: '',
    taskType: 0,
    updateSource: true,
    version: 0,
    whereToEnter: 0,
    createUserId: 0,
    projectId: 0,
    tenantId: 0,
    userId: 0
}
