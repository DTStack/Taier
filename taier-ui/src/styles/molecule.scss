@import "./variables.scss";

.mo-menuBar--horizontal {
    height: $menuBar-size;
    background: var(--activityBar-background);

    > .mo-menu {
        height: $menuBar-size;
        color: var(--activityBar-inactiveForeground);
        background: inherit;
    }
}

// only the tabs in editor should have hover styles
.mo-editor__group {
    .mo-tab__item:hover {
        background-color: var(--tab-hoverBackground);
    }
}

.mo-tab__item__status,
.mo-tab__item__op {
    color: inherit;

    &:hover {
        color: inherit;
    }
}

.mo-editor__group-breadcrumb {
    color: var(--breadcrumb-foreground);

    .mo-breadcrumb__item:hover {
        color: var(--breadcrumb-focusForeground);
        opacity: 1;
    }
}

.mo-notification__item:focus {
    background: var(--list-hoverBackground);
    outline: 1px solid var(--list-focusOutline);
}

.mo-statusBar__item > a:empty {
    padding: 0;
}

// ===================== Auxiliary Bar ===========================
.mo-auxiliaryBar__tabs {
    .mo-auxiliaryBar__tabs__tab {
        font-size: 12px;
        padding-top: 5px;
        padding-bottom: 5px;

        &:hover {
            color: var(--taier-rightBar-foreground);
            background-color: var(--taier-rightBar-background);
        }

        &--active {
            color: var(--taier-rightBar-foreground);
            background-color: var(--taier-rightBar-background);
            position: relative;

            &::after {
                position: absolute;
                top: 0;
                right: 0;
                bottom: 0;
                width: 2px;
                height: 100%;
                background-color: #3f87ff;
                content: "";
            }
        }
    }
}

.mo-auxiliaryBar__container {
    background: var(--sideBar-background);
}
