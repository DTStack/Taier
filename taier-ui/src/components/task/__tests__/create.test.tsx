import { selectItem, toggleOpen } from '@/tests/utils';
import molecule from '@dtinsight/molecule';
import { cleanup, fireEvent, render, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { act } from 'react-dom/test-utils';
import Create from '../create';
import type { IOfflineTaskProps } from '@/interface';
import api from '@/api';
import { Form, Input } from 'antd';

jest.useFakeTimers();
jest.mock('@/api');

(molecule.folderTree.getState as jest.Mock).mockReset().mockImplementation(() => ({
    folderTree: {
        data: [{ id: 1 }],
    },
}));

jest.mock('@/services', () => ({
    taskRenderService: {
        renderCreateForm: jest.fn(() => (
            <>
                <Form.Item name="syncModel">
                    <Input data-testid="syncModel" />
                </Form.Item>
                <Form.Item name="test">
                    <Input data-testid="test" />
                </Form.Item>
                <Form.Item name="pythonVersion">
                    <Input data-testid="test" />
                </Form.Item>
            </>
        )),
        getState: jest.fn(() => ({
            supportTaskList: [
                {
                    key: 1,
                    taskProperties: {
                        formField: ['syncModel', 'pythonVersion', 'test'],
                    },
                },
            ],
        })),
    },
}));

jest.mock('@/context', () => {
    const react = jest.requireActual('react');
    return react.createContext({
        supportJobTypes: [{ key: 1, value: 'spark' }],
    });
});

describe('Test Create Component', () => {
    beforeEach(() => {
        cleanup();
    });
    it('Should match snapshot', () => {
        const { asFragment } = render(<Create />);
        expect(asFragment()).toMatchSnapshot();
    });

    describe('Test Create Without Record', () => {
        it('Should support create task', async () => {
            (molecule.editor.updateTab as jest.Mock).mockReset();
            const fn = jest.fn().mockResolvedValue(true);
            const { container } = render(<Create current={{ id: 'test', tab: { id: 1, data: {} } }} onSubmit={fn} />);

            // Edit name
            fireEvent.change(container.querySelector('input#name')!, { target: { value: 'test' } });
            expect(molecule.editor.updateTab).lastCalledWith({
                data: {
                    name: 'test',
                    nodePid: 1,
                    sqlText: undefined,
                    taskDesc: undefined,
                    taskType: undefined,
                },
                id: 1,
                status: 'edited',
            });

            // Select taskType
            toggleOpen(container);
            selectItem(0);
            expect(molecule.editor.updateTab).lastCalledWith({
                data: {
                    name: 'test',
                    nodePid: 1,
                    sqlText: undefined,
                    taskDesc: undefined,
                    taskType: 1,
                },
                id: 1,
                status: 'edited',
            });

            act(() => {
                // submit
                fireEvent.click(container.querySelector('button.ant-btn[type="submit"]')!);
            });

            await waitFor(() => {
                expect(fn).toBeCalledTimes(1);
                expect(fn).toBeCalledWith({
                    name: 'test',
                    nodePid: 1,
                    sqlText: undefined,
                    taskDesc: undefined,
                    taskType: 1,
                });
            });
        });

        it('Should log error on name field', async () => {
            (molecule.editor.updateTab as jest.Mock).mockReset();
            const { container, getByText } = render(<Create current={{ id: 'test', tab: { id: 1, data: {} } }} />);

            // submit
            fireEvent.click(container.querySelector('button.ant-btn[type="submit"]')!);

            await waitFor(() => {
                expect(getByText('任务名称不可为空！')).toBeInTheDocument();
                expect(
                    getByText('请选择任务类型', {
                        selector: 'div',
                    })
                ).toBeInTheDocument();
            });

            // Edit name
            fireEvent.change(container.querySelector('input#name')!, {
                target: { value: new Array(200).fill('a').join('') },
            });
            // submit
            fireEvent.click(container.querySelector('button.ant-btn[type="submit"]')!);

            await waitFor(() => {
                expect(getByText('任务名称不得超过128个字符！')).toBeInTheDocument();
            });

            // Edit name
            fireEvent.change(container.querySelector('input#name')!, {
                target: { value: 'a-b%' },
            });
            // submit
            fireEvent.click(container.querySelector('button.ant-btn[type="submit"]')!);

            await waitFor(() => {
                expect(getByText('任务名称只能由字母、数字、中文、下划线组成!')).toBeInTheDocument();
            });
        });

        it('Should log errors on taskDesc field', async () => {
            (molecule.editor.updateTab as jest.Mock).mockReset();
            const { container, getByText } = render(<Create current={{ id: 'test', tab: { id: 1, data: {} } }} />);

            // Edit taskDesc
            fireEvent.change(container.querySelector('textarea#taskDesc')!, {
                target: { value: new Array(300).fill('a').join('') },
            });

            // submit
            fireEvent.click(container.querySelector('button.ant-btn[type="submit"]')!);

            await waitFor(() => {
                expect(getByText('描述请控制在200个字符以内！')).toBeInTheDocument();
            });
        });
    });

    describe('Test Create With Record', () => {
        it('Should alert tips about version updated', async () => {
            (api.getOfflineTaskByID as jest.Mock).mockReset().mockResolvedValue({
                code: 1,
                data: {
                    syncModel: 1,
                },
            });
            const { getByText } = render(
                <Create current={{ id: 'test', tab: { id: 1, data: {} } }} record={{ id: 1 } as IOfflineTaskProps} />
            );

            await waitFor(() => {
                expect(getByText('由于版本更新，需要重新设置该字段')).toBeInTheDocument();
            });
        });

        it('Should set value into fields', async () => {
            (api.getOfflineTaskByID as jest.Mock).mockReset().mockResolvedValue({
                code: 1,
                data: {
                    name: 'test',
                    taskType: 1,
                    taskDesc: 'abc',
                    sqlText: 'aaa',
                    sourceMap: {
                        syncModal: 1,
                    },
                    exeArgs: JSON.stringify({ '--app-type': 'python3' }),
                    test: 1,
                },
            });
            const fn = jest.fn().mockResolvedValue(true);
            const { container } = render(
                <Create
                    current={{ id: 'test', tab: { id: 1, data: {} } }}
                    record={{ id: 1 } as IOfflineTaskProps}
                    onSubmit={fn}
                />
            );

            await waitFor(() => {
                expect(container.querySelector<HTMLInputElement>('input#name')?.value).toBe('test');
            });

            act(() => {
                // submit
                fireEvent.click(container.querySelector('button.ant-btn[type="submit"]')!);
            });

            await waitFor(() => {
                expect(fn).toBeCalledWith({
                    name: 'test',
                    nodePid: 1,
                    sqlText: 'aaa',
                    syncModel: undefined,
                    taskDesc: 'abc',
                    taskType: 1,
                    pythonVersion: 3,
                    test: undefined,
                });
            });
        });

        it('Should fill fields with record', async () => {
            const fn = jest.fn().mockResolvedValue(true);
            const { container } = render(
                <Create
                    current={{ id: 'test', tab: { id: 1, data: {} } }}
                    record={
                        {
                            id: 1,
                            name: 'test',
                            taskType: 1,
                            taskDesc: 'abc',
                            sqlText: 'aaa',
                            sourceMap: {
                                syncModal: 1,
                            },
                            exeArgs: JSON.stringify({ '--app-type': 'python3' }),
                            test: 1,
                        } as any
                    }
                    isRequest={false}
                    onSubmit={fn}
                />
            );

            await waitFor(() => {
                expect(container.querySelector<HTMLInputElement>('input#name')?.value).toBe('test');
            });

            act(() => {
                // submit
                fireEvent.click(container.querySelector('button.ant-btn[type="submit"]')!);
            });

            await waitFor(() => {
                expect(fn).toBeCalledWith({
                    name: 'test',
                    nodePid: 1,
                    sqlText: 'aaa',
                    syncModel: undefined,
                    taskDesc: 'abc',
                    taskType: 1,
                    pythonVersion: undefined,
                    test: 1,
                });
            });
        });
    });
});
