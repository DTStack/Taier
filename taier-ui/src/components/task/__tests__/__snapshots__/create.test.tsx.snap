// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Create Component Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="taier__create__container"
  >
    <div
      class="taier__create__wrapper"
    >
      <div
        class="ant-spin-nested-loading"
      >
        <div
          class="ant-spin-container"
        >
          <form
            autocomplete="off"
            class="ant-form ant-form-vertical taier__create__form"
          >
            <div
              class="ant-form-item"
            >
              <div
                class="ant-row ant-form-item-row"
              >
                <div
                  class="ant-col ant-form-item-label"
                >
                  <label
                    class="ant-form-item-required"
                    for="name"
                    title="任务名称"
                  >
                    任务名称
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        aria-required="true"
                        class="ant-input"
                        id="name"
                        placeholder="请输入任务名称"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-form-item"
            >
              <div
                class="ant-row ant-form-item-row"
              >
                <div
                  class="ant-col ant-form-item-label"
                >
                  <label
                    class="ant-form-item-required"
                    for="taskType"
                    title="任务类型"
                  >
                    任务类型
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <div
                        aria-required="true"
                        class="ant-select ant-select-in-form-item ant-select-single ant-select-show-arrow ant-select-show-search"
                      >
                        <div
                          class="ant-select-selector"
                        >
                          <span
                            class="ant-select-selection-search"
                          >
                            <input
                              aria-activedescendant="taskType_list_0"
                              aria-autocomplete="list"
                              aria-controls="taskType_list"
                              aria-haspopup="listbox"
                              aria-owns="taskType_list"
                              aria-required="true"
                              autocomplete="off"
                              class="ant-select-selection-search-input"
                              id="taskType"
                              role="combobox"
                              type="search"
                              value=""
                            />
                          </span>
                          <span
                            class="ant-select-selection-placeholder"
                          >
                            请选择任务类型
                          </span>
                        </div>
                        <span
                          aria-hidden="true"
                          class="ant-select-arrow"
                          style="user-select: none;"
                          unselectable="on"
                        >
                          <span
                            aria-label="down"
                            class="anticon anticon-down ant-select-suffix"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="down"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-form-item"
            >
              <div
                class="ant-row ant-form-item-row"
              >
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        class="ant-input"
                        data-testid="syncModel"
                        id="syncModel"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-form-item"
            >
              <div
                class="ant-row ant-form-item-row"
              >
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        class="ant-input"
                        data-testid="test"
                        id="test"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-form-item"
            >
              <div
                class="ant-row ant-form-item-row"
              >
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        class="ant-input"
                        data-testid="test"
                        id="pythonVersion"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-form-item"
            >
              <div
                class="ant-row ant-form-item-row"
              >
                <div
                  class="ant-col ant-form-item-label"
                >
                  <label
                    class="ant-form-item-required"
                    for="nodePid"
                    title="存储位置"
                  >
                    存储位置
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        class="ant-input"
                        type="hidden"
                        value="1"
                      />
                      <div
                        aria-required="true"
                        class="ant-select ant-tree-select ant-select-in-form-item ant-select-single ant-select-show-arrow"
                      >
                        <div
                          class="ant-select-selector"
                        >
                          <span
                            class="ant-select-selection-search"
                          >
                            <input
                              aria-autocomplete="list"
                              aria-controls="nodePid_list"
                              aria-haspopup="listbox"
                              aria-owns="nodePid_list"
                              aria-required="true"
                              autocomplete="off"
                              class="ant-select-selection-search-input"
                              id="nodePid"
                              readonly=""
                              role="combobox"
                              style="opacity: 0;"
                              type="search"
                              unselectable="on"
                              value=""
                            />
                          </span>
                          <span
                            class="ant-select-selection-item"
                            title="1"
                          >
                            1
                          </span>
                        </div>
                        <span
                          aria-hidden="true"
                          class="ant-select-arrow"
                          style="user-select: none;"
                          unselectable="on"
                        >
                          <span
                            aria-label="down"
                            class="anticon anticon-down ant-select-suffix"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="down"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-form-item"
            >
              <div
                class="ant-row ant-form-item-row"
              >
                <div
                  class="ant-col ant-form-item-label"
                >
                  <label
                    class=""
                    for="taskDesc"
                    title="描述"
                  >
                    描述
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <textarea
                        class="ant-input"
                        id="taskDesc"
                        placeholder="请输入描述"
                        rows="4"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-form-item ant-form-item-hidden"
            >
              <div
                class="ant-row ant-form-item-row"
              >
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-form-item"
            >
              <div
                class="ant-row ant-form-item-row"
              >
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <button
                        class="ant-btn ant-btn-primary"
                        type="submit"
                      >
                        <span>
                          确 认
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
